frontend/scripts/containers/shared/search-react.container.js

```diff
@@ -0,0 +1,49 @@
+import { bindActionCreators } from 'redux';
+import { connect } from 'preact-redux';
+import groupBy from 'lodash/groupBy';
+import Search from 'react-components/shared/search.component.js';
+import { selectExpandedNode, selectNode } from 'actions/tool.actions';
+import { ACTORS_COLUMN_IDS, IMPORTER_EXPORTER_TYPE } from 'constants';
+
+let searchNodes;
+
+const mapStateToProps = (state) => {
+  const { nodes, selectedNodesIds } = state.tool;
+  // store nodes at container level to avoid rerendering when filtering... for want of a better solution
+  if (nodes !== undefined && (!searchNodes || nodes.length !== searchNodes.length)) {
+    const allNodes = nodes.filter(
+      node => node.hasFlows === true &&
+      node.isAggregated !== true &&
+      node.isUnknown !== true
+    );
+
+    searchNodes = Object.values(groupBy(allNodes.filter(x => ACTORS_COLUMN_IDS.includes(x.columnId)), 'mainNodeId'))
+      .map(([nA, nB]) => nB ?
+        ({
+          id: `${nA.id}_${nB.id}`,
+          name: nA.name,
+          type: IMPORTER_EXPORTER_TYPE,
```

Comment:
We can also remove this (also evil) constant if we group nodes by main_node_id instead of looking for specific patterns based on hardcoded, unflexible constants

Reply:
The `mainNodeId` is already used for the grouping. This is just a string used for reflecting nodes as grouped in the result list. And to filter the results that have a link to profile pages.

Reply:
I know. What I'm saying is that, instead of using a set string, we should compute it. Imagine that, in a few months, we get a "shipping company" column that has the same actors. We would need to create constants for every combination of it. bad for us

Reply:
true true, thanks I'll change it.